astropy/wcs/src/unit_list_proxy.c

```diff
@@ -186,6 +187,46 @@ PyUnitListProxy_getitem(
   return result;
 }
 
+static PyObject*
+PyUnitListProxy_richcmp(
+	PyObject *a,
+	PyObject *b,
+	int op){
+  PyUnitListProxy *lhs, *rhs;
+  assert(a != NULL && b != NULL);
+  if (!PyObject_TypeCheck(a, &PyUnitListProxyType) || 
+  	  !PyObject_TypeCheck(b, &PyUnitListProxyType)) {
+  	Py_RETURN_NOTIMPLEMENTED;
+  }
+  if (op != Py_EQ && op != Py_NE) {
+  	Py_RETURN_NOTIMPLEMENTED;
+  }
+  lhs = (PyUnitListProxy *)a;
+  rhs = (PyUnitListProxy *)b;
+  int equal = PyObject_RichCompareBool(lhs->unit_class, rhs->unit_class, Py_EQ);
+  if (equal == -1) {
+    PyErr_SetString(PyExc_TypeError,
```

Comment:
`PyObject_RichCompareBool` already sets an exception when it returns -1, so this will lead to a memory leak (and a worse exception type and message). Maybe a good idea to keep the inline comments I made in https://github.com/astropy/astropy/pull/8480#issuecomment-474552914